components/eamxx/src/share/io/eamxx_output_manager.cpp

```diff
@@ -72,13 +62,27 @@ setup (const std::map<std::string,std::shared_ptr<fm_type>>& field_mgrs,
   }
 
   // For each grid, create a separate output stream.
-  if (field_mgrs.size()==1) {
-    auto output = std::make_shared<output_type>(m_io_comm,m_params,field_mgrs.begin()->second,grids_mgr);
+  if (m_params.isParameter("Field Names")) {
```

Comment:
Btw, I think this version of the parameter list is only used in unit tests. I'd be in favor of always forcing to specify the general (nested) format: `FIelds-><GRIDNAME>->Field Names`.  We can of course do it in another PR, but I felt like bringing it up here for discussion, since you had to add a fix for this.

Reply:
I agree, let me add a todo